Counterexamples/Phillips.lean

```diff
@@ -0,0 +1,652 @@
+`[0, 1]` (or rather an arbitrary extension of Lebesgue integration to all bounded functions,
+thanks to Hahn-Banach). Then `φ g` should be the total mass of the continuous part of `φ`,
+which is `1`. This contradicts the fact that `g = 0`, and concludes the proof that `f` has no
+Pettis integral.
+
+## Implementation notes
+
+The space of all bounded functions is defined as the space of all bounded continuous functions
+on a discrete copy of the original type, as mathlib only contains the space of all bounded
+continuous functions (which is the useful one).
+-/
+
+
+namespace Counterexample
+
+universe u
+
+variable {α : Type u}
+
+open Set BoundedContinuousFunction MeasureTheory
+
+open Cardinal (aleph)
+
+open scoped Cardinal BoundedContinuousFunction
+
+noncomputable section
+
+/-- A copy of a type, endowed with the discrete topology -/
+def DiscreteCopy (α : Type u) : Type u :=
+  α
+#align counterexample.discrete_copy Counterexample.DiscreteCopy
+
+instance : TopologicalSpace (DiscreteCopy α) :=
+  ⊥
+
+instance : DiscreteTopology (DiscreteCopy α) :=
+  ⟨rfl⟩
+
+instance [Inhabited α] : Inhabited (DiscreteCopy α) :=
+  ⟨show α from default⟩
+
+namespace Phillips1940
+
+/-!
+### Extending the integral
+
+Thanks to Hahn-Banach, one can define a (non-canonical) continuous linear functional on the space
+of all bounded functions, coinciding with the integral on the integrable ones.
+-/
+
+attribute [-instance] Real.pseudoMetricSpace
+
+/-- The subspace of integrable functions in the space of all bounded functions on a type.
+This is a technical device, used to apply Hahn-Banach theorem to construct an extension of the
+integral to all bounded functions. -/
+def boundedIntegrableFunctions [MeasurableSpace α] (μ : Measure α) :
+    Subspace ℝ (DiscreteCopy α →ᵇ ℝ) where
+  carrier := {f | Integrable f μ}
+  zero_mem' := integrable_zero _ _ _
+  add_mem' hf hg := Integrable.add hf hg
+  smul_mem' c _ hf := Integrable.smul c hf
+#align counterexample.phillips_1940.bounded_integrable_functions Counterexample.Phillips1940.boundedIntegrableFunctions
+
+/-- The integral, as a continuous linear map on the subspace of integrable functions in the space
+of all bounded functions on a type. This is a technical device, that we will extend through
+Hahn-Banach. -/
+def boundedIntegrableFunctionsIntegralCLM [MeasurableSpace α] (μ : Measure α) [IsFiniteMeasure μ] :
+    boundedIntegrableFunctions μ →L[ℝ] ℝ :=
+  LinearMap.mkContinuous (F := ℝ)
+    { toFun := fun f => ∫ x, f.1 x ∂μ
+      map_add' := fun f g => integral_add f.2 g.2
+      map_smul' := fun c f => integral_smul c f.1 } (μ univ).toReal
+    (by
+      intro f
+      rw [mul_comm]
+      apply norm_integral_le_of_norm_le_const
+      apply Filter.eventually_of_forall
+      intro x
+      exact BoundedContinuousFunction.norm_coe_le_norm f.1 x)
+#align counterexample.phillips_1940.bounded_integrable_functions_integral_clm Counterexample.Phillips1940.boundedIntegrableFunctionsIntegralCLM
```

Comment:
Without removing the instance `Real.pseudoMetricSpace`, `boundedIntegrableFunctionsIntegralCLM` can't be defined.
Is this a counterattack of reenable eta!? 😧😰😱

Reply:
Could you start a zulip thread about this one? I don't see what's wrong, but it does seem worrying. :-(

Reply:
```lean
def boundedIntegrableFunctionsIntegralCLM [MeasurableSpace α] (μ : Measure α) [IsFiniteMeasure μ] :
    boundedIntegrableFunctions μ →L[ℝ] ℝ :=
  LinearMap.mkContinuous (E := ↥(boundedIntegrableFunctions μ))
    { toFun := fun f => ∫ x, f.1 x ∂μ
      map_add' := fun f g => integral_add f.2 g.2
      map_smul' := fun c f => integral_smul c f.1 } (μ univ).toReal
    (by
      intro f
      rw [mul_comm]
      apply norm_integral_le_of_norm_le_const
      apply Filter.eventually_of_forall
      intro x
      exact BoundedContinuousFunction.norm_coe_le_norm f.1 x)
```
I solved the problem by adding type ascription, it's likely just a type inference problem. 😅
R.I.P reenable eta 👻